ext/spl/spl_iterators.c

```diff
@@ -2793,12 +2723,12 @@ SPL_METHOD(CachingIterator, offsetUnset)
 
 	SPL_FETCH_AND_CHECK_DUAL_IT(intern, ZEND_THIS);
 
-	if (!(intern->u.caching.flags & CIT_FULL_CACHE))	{
-		zend_throw_exception_ex(spl_ce_BadMethodCallException, 0, "%s does not use a full cache (see CachingIterator::__construct)", ZSTR_VAL(Z_OBJCE_P(ZEND_THIS)->name));
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "S", &key) == FAILURE) {
```

Comment:
Dam, SPL was a mess. Thanks for taking care of it 👏🏻

Reply:
Yeah, not only messy, but really big, too... It's the worst.